src/box/alter.cc

```diff
@@ -1095,11 +1142,7 @@ static inline void
 space_check_format_with_yield(struct space *space,
 			      struct tuple_format *format)
 {
-	struct txn *txn = in_txn();
-	assert(txn != NULL);
-	(void) txn_can_yield(txn, true);
-	auto yield_guard =
-		make_scoped_guard([=] { txn_can_yield(txn, false); });
+	AlterYieldGuard guard(in_txn(), space);
```

Comment:
AlterYieldGuard is only taken by `space:format` and `index:alter`. `AlterSpaceLock` covers the cases when, for example, someone is trying to rename a space during `index:alter`.

Reply:
Wrote about it in commit message.

Reply:
Maybe, better modify the existing DDL lock taken for each DDL statement so that it's released it only after the statement is committed or rolled back, like `AlterYieldGuard`? This would increase the number of conflicts but I think that should be fine because I doubt anyone expects that modifying the same space from different fibers will work. Would it break any tests?

Reply:
Discussed f2f, decided to release lock after commit/rollback.
Done.